core/src/test/java/org/apache/iceberg/TestPaths.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import java.net.URISyntaxException;
+import org.apache.hadoop.fs.Path;
+import org.junit.Test;
+
+public class TestPaths {
+
+  @Test
+  public void testPathsWithNoTrailingSlash() {
+    String pathWithoutTrailingSlash = "s3a://bucket1";
+    String metadataLocation = "/metadata";
+    String metadataLocationWithoutLeadingSlash = "metadata";
+
+    String expectedPath = "s3a://bucket1/metadata";
+
+    assertThrows(
```

Comment:
better to use AssertJ assertions here: `Assertions.assertThatThrownBy(() -> ...).isInstanceOf(..).hasMessage(..)`

Reply:
Done.